connectors/v2/youtube.js

```diff
@@ -82,28 +82,35 @@ var timestampRegex = new RegExp(timestampPattern,'gi');
 // More to the point: the playlist only changes when the YT page does, so no point constantly scanning it.
 Connector.setThrottleInterval(3000);
 
-// The playlist does not change on its own, so cache it and invalidate only when video id changes.
+// The playlist does not change on its own, so cache it and invalidate when the video id or DOM changes
 var playlistCache = {
 	hash: null,
-	value: null
+	value: null,
+	getPageHash: function () {
+		return Connector.getUniqueID() + $('#eow-description, .comment-text-content').length;
+	},
+	isValid: function () {
+		return (this.hash == this.getPageHash());
+	},
+	setValue: function(newValue) {
+		this.hash = this.getPageHash();
+		this.value = newValue;
+	}
 };
```

Comment:
Is this something to consider putting in BaseConnector logic, rather than individual connectors? You'd imagine it's the same for all tracks - playlists don't change, as a rule.